cdap-data-fabric-tests/src/test/java/co/cask/cdap/data/stream/service/DFSStreamHeartbeatsTest.java

```diff
@@ -0,0 +1,233 @@
+import co.cask.cdap.data2.transaction.stream.StreamConsumerStateStoreFactory;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamConsumerStateStoreFactory;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileAdmin;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileConsumerFactory;
+import co.cask.cdap.explore.guice.ExploreClientModule;
+import co.cask.cdap.gateway.auth.AuthModule;
+import co.cask.tephra.TransactionManager;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.util.Modules;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.twill.internal.zookeeper.InMemoryZKServer;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public class DFSStreamHeartbeatsTest {
+  private static final String API_KEY = "SampleTestApiKey";
+  private static final byte[] TWO_BYTES = new byte[] { 'a', 'b' };
+
+  private static final String HOSTNAME = "127.0.0.1";
+  private static int port;
+  private static CConfiguration conf;
+
+  private static Injector injector;
+  private static StreamHttpService streamHttpService;
+  private static StreamService streamService;
+  private static TransactionManager txService;
+  private static DatasetOpExecutor dsOpService;
+  private static DatasetService datasetService;
+  private static MockHeartbeatPublisher heartbeatPublisher;
+  private static InMemoryZKServer zkServer;
+  private static ZKClientService zkClient;
+
+  @ClassRule
+  public static TemporaryFolder tmpFolder = new TemporaryFolder();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    zkServer = InMemoryZKServer.builder().setDataDir(tmpFolder.newFolder()).build();
+    zkServer.startAndWait();
+
+    conf = CConfiguration.create();
+    conf.set(Constants.Zookeeper.QUORUM, zkServer.getConnectionStr());
+    conf.setInt(Constants.Stream.CONTAINER_INSTANCE_ID, 0);
+
+    conf.set(Constants.Router.ADDRESS, HOSTNAME);
+    conf.setInt(Constants.Router.ROUTER_PORT, 0);
+    conf.set(Constants.CFG_LOCAL_DATA_DIR, tmpFolder.newFolder().getAbsolutePath());
+    injector = Guice.createInjector(
```

Comment:
I do need to start an HTTP service to have the StreamHandler running, since it participates in the logic that I test. Also, the DistributedStreamService needs a whole lot of other things and I thing it's easier to set them via injection. I will reduce the size of the injections by creating an ``InMemoryStreamMetastore``.